backend/mark_marker/src/main/scala/mark_marker/Utils.scala

```diff
@@ -0,0 +1,28 @@
+package mark_marker
+
+import java.io.File
+
+import edu.stanford.nlp.classify.{Classifier, ColumnDataClassifier}
+import mark_marker.trainer.DatabaseConnector
+
+import scala.io.Source
+
+/**
+  * Created by eb1314 on 27/11/16.
+  */
+object Utils {
+
+  def matchType(t: String) = t match {
+    case s if s.contains("haskell") => ".hs"
+    case s if s.contains("java") => ".java"
+    case _ => throw new IllegalArgumentException("Invalid language")
```

Comment:
If we go with the "it's a type, not a language" rhetoric, you should report invalid type i guess?